src/components/app/Home.js

```diff
@@ -266,8 +266,19 @@ class Home extends React.PureComponent<HomeProps, HomeState> {
     if (!event.dataTransfer) {
       return;
     }
-
+    // console.log(event.dataTransfer);
+    const firstItem = event.dataTransfer.items[0];
+
+    if ('webkitGetAsEntry' in firstItem) {
+      const webkitEntry = firstItem.webkitGetAsEntry();
+      // console.log(webkitEntry);
+      if (webkitEntry.isDirectory && webkitEntry.name.endsWith('.trace')) {
+        this.props.retrieveProfileFromFile(webkitEntry);
+        return;
+      }
+    }
```

Comment:
Since we destruct the `event.dataTransfer` below, we can also get the `items` from that line like this and move the upper if after that line as well:
`const { files, items } = event.dataTransfer;`
or better, we can get the first item directly like this: 
`const { files, items: [firstItem] } = event.dataTransfer;`

Reply:
Oh great! I didn't know about the second syntax.